implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheusmetrics/PrometheusNamingConvention.java

```diff
@@ -85,7 +85,7 @@ else if (!conventionName.endsWith("_seconds")) {
      */
     @Override
     public String tagKey(String key) {
-        return PrometheusNaming.sanitizeLabelName(key);
+        return PrometheusNaming.sanitizeLabelName(PrometheusNaming.prometheusName(key));
```

Comment:
This is the change that makes label names consistent to metric names inside of the registry and for users who directly want to use this `NamingConvention`.

Reply:
I wonder why we need to do this since we're already calling the Prometheus library's `sanitizeLabelName`. I would expect that to do any needed transformation of a label name. I wonder if its current behavior is intentional or not. It seems possible that there could be subtle intended differences in the rules about metric names vs labels, so using both here could be problematic in the future if it isn't already.

Reply:
For the record: we asked @fstab, for now we should do `prometheusName` + `sanitizeXyz`.